### PR TITLE
Pac 38 orika

### PR DESCRIPTION
**Pull request by tomoicki**
Renamed and refactored to OrikaMapper. Left old mapper cuz i wasnt sure about its deletion.
Please review. If it still needs any changes please let me know.

### CODE REVIEW

File: backend/src/main/java/com/webappsbusters/parcelmanagement/controller/ParcelController.java [hunk 1]
```diff
@@ -16,18 +17,20 @@
 
     private final ParcelService parcelService;
     private final ParcelMapper parcelMapper;
```
Comment (TheFlyingKeyboard) [inline backend/src/main/java/com/webappsbusters/parcelmanagement/controller/ParcelController.java]: I think we should remove this old mapper, as it is not needed now. Remove ParcelMapper.class and this field

File: backend/src/main/java/com/webappsbusters/parcelmanagement/mapper/OrikaAttemptTest.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+public class OrikaAttemptTest {
+    public static void main(String[] args) {
+        //Parcel parcel1 = new Parcel(ParcelStatus.OUT_FOR_DELIVERY,"sendcity","sendpost","sendstreet","recivcity","receivpost","receivstreet",100,10,20,30,"L");
+        Parcel parcel1 = new Parcel.ParcelBuilder().receiverCity("wroclaw").build();
+        MapperFacade facade = new OrikaMapper();
+        ParcelDto parcel1clone = facade.map(parcel1, ParcelDto.class);
+        System.out.println("origin: " + parcel1);
+        System.out.println("clone: " + parcel1clone);
+    }
```
Comment (TheFlyingKeyboard) [inline backend/src/main/java/com/webappsbusters/parcelmanagement/mapper/OrikaAttemptTest.java]: I think we don't need this class, as we know the new mapper works

File: backend/src/main/java/com/webappsbusters/parcelmanagement/controller/ParcelController.java [hunk 3]
```diff
@@ -16,18 +17,20 @@
...
+        this.mapperFacade = mapperFacade;
     }
 
     @GetMapping(value = "/parcels/{parcelId}")
     public ResponseEntity<ParcelDto> getParcel(@PathVariable String parcelId) {
         ParcelDto parcelDto =
-                parcelMapper.mapToParcelDto(parcelService.getParcelById(parcelId).
-                        orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND)));
+                mapperFacade.map(parcelService.getParcelById(parcelId).
+                        orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND)),ParcelDto.class);
```
Comment (TheFlyingKeyboard) [inline backend/src/main/java/com/webappsbusters/parcelmanagement/controller/ParcelController.java]: please reformat - add space after  ","

File: backend/pom.xml [hunk 4]
```diff
@@ -31,7 +31,11 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-web</artifactId>
 		</dependency>
-
+		<dependency>
+			<groupId>ma.glasnost.orika</groupId>
+			<artifactId>orika-core</artifactId>
+			<version>1.5.4</version>
```
Comment (TheFlyingKeyboard) [inline backend/pom.xml]: Please extract version to properties - intellij can do that, it is in right click -> refactor -> extract property or introduce property

**Review by TheFlyingKeyboard — CHANGES_REQUESTED** [review]

**Review by TheFlyingKeyboard — APPROVED** [review]

**Comment by tomoicki** [comment]
Można już zrobić "Merge pull request" czy coś posuje?

**Review by joakrzem — APPROVED** [review]

**Pull request merged**